server/src/main/java/org/elasticsearch/common/util/Maps.java

```diff
@@ -328,41 +328,4 @@ public static <K, V1, V2> Map<K, V2> transformValues(Map<K, V1> source, Function
         }
         return copy;
     }
-
-    /**
-     * An immutable implementation of {@link Map.Entry}.
-     * @param key key
-     * @param value value
-     */
-    public record ImmutableEntry<KType, VType>(KType key, VType value) implements Map.Entry<KType, VType> {
```

Comment:
Just realized the default one does not permit null key/values. Was it the point for this class?

Reply:
Yes, I believe that was the intention

https://github.com/elastic/elasticsearch/pull/88815#discussion_r930099929

Reply:
Updated the javadoc to document this difference.